src/services/shims.ts

```diff
@@ -485,18 +485,24 @@ namespace ts {
 
     export class CoreServicesShimHostAdapter implements ParseConfigHost, ModuleResolutionHost, JsTyping.TypingResolutionHost {
 
-        public directoryExists: (directoryName: string) => boolean;
+        public directoryExists: ((directoryName: string) => boolean);
         public realpath: (path: string) => string;
         public useCaseSensitiveFileNames: boolean;
 
         constructor(private shimHost: CoreServicesShimHost) {
-        this.useCaseSensitiveFileNames = this.shimHost.useCaseSensitiveFileNames ? this.shimHost.useCaseSensitiveFileNames() : false;
+            this.useCaseSensitiveFileNames = this.shimHost.useCaseSensitiveFileNames ? this.shimHost.useCaseSensitiveFileNames() : false;
             if ("directoryExists" in this.shimHost) {
                 this.directoryExists = directoryName => this.shimHost.directoryExists(directoryName);
             }
+            else {

```

Comment:
why?

Reply:
`TypingResolutionHost` requires `directoryExists` to be non-optional. It's likely that `this.shimHost` always contains `directoryExists`, so it always gets defined.